src/components/enterprise-banner/OffersAlert.jsx

```diff
@@ -1,62 +1,29 @@
-import React from 'react';
-import PropTypes from 'prop-types';
-import { connect } from 'react-redux';
+import React, { useContext } from 'react';
+
 import { Alert } from '@edx/paragon';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import { faInfoCircle } from '@fortawesome/free-solid-svg-icons';
-
-import { fetchOffers } from '../dashboard/sidebar/offers';
+import { UserSubsidyContext } from '../enterprise-user-subsidy';
 
 export const getOffersText = (number) => `You have ${number} course redemption voucher${number > 1 ? 's' : ''} left to use.`;
 
-export class BaseOffersAlert extends React.Component {
-  componentDidMount() {
-    const { isOffersLoading, fetchOffersAction } = this.props;
-
-    if (!isOffersLoading) {
-      fetchOffersAction('full_discount_only=True');
-    }
-  }
-
-  render() {
-    const { offersCount } = this.props;
-    if (offersCount) {
-      return (
-        <Alert
-          className="pl-5"
-          variant="info"
-        >
-          <div className="container">
-            <FontAwesomeIcon className="mr-1" icon={faInfoCircle} />
-            {getOffersText(offersCount)}
-          </div>
-        </Alert>
-      );
-    }
-    return null;
+const OffersAlert = () => {
+  const { offers } = useContext(UserSubsidyContext);
+
+  if (offers.offersCount) {
+    return (
+      <Alert
+        className="pl-5"
+        variant="info"
+      >
+        <div className="container">
+          <FontAwesomeIcon className="mr-1" icon={faInfoCircle} />
+          {getOffersText(offers.offersCount)}
+        </div>
+      </Alert>
+    );
   }
-}
-
-BaseOffersAlert.defaultProps = {
-  fetchOffersAction: null,
-  isOffersLoading: false,
-  offersCount: 0,
-};
-
-BaseOffersAlert.propTypes = {
-  fetchOffersAction: PropTypes.func,
-  isOffersLoading: PropTypes.bool,
-  offersCount: PropTypes.number,
+  return null;
```

Comment:
nit: might make sense to return earlier in the function, e.g. `if (!offers.offersCount) { return null; }`

Reply:
I'm gonna leave this one because I don't think it makes a huge difference here and it's not in scope for the refactor.

Reply:
It's just a pattern that results in cleaner code; we should try to always prefer to return early where relevant. e.g., see https://medium.com/swlh/return-early-pattern-3d18a41bba8#:~:text=Return%20early%20is%20the%20way,when%20conditions%20are%20not%20met.